src/main/java/seedu/address/model/ResumeBook.java

```diff
@@ -357,7 +372,14 @@ public void deleteSkill(Skill key) {
         skills.remove(key);
         for (Item item : resumes) {
             Resume resume = (Resume) item;
-            resume.getSkillIds().remove(Integer.valueOf(id));
+            Resume newresume = new Resume(resume.getName(), resume.getId(), resume.getTags());
+            newresume.setInternshipIds(resume.getInternshipIds());
+            newresume.setProjectIds(resume.getProjectIds());
+            List<Integer> newSkillList = new ArrayList<>();
+            newSkillList.addAll(resume.getSkillIds());
+            newresume.setSkillIds(newSkillList);
+            newresume.getSkillIds().remove(Integer.valueOf(id));
+            setResume(resume, newresume);
```

Comment:
Making it consistent with internship
```suggestion
            Resume newResume = new Resume(resume.getName(), resume.getId(), resume.getTags());
            List<Integer> newSkillList = new ArrayList<>();
            newSkillList.addAll(resume.getSkillIds());
            
            newResume.setInternshipIds(resume.getInternshipIds());
            newResume.setProjectIds(resume.getProjectIds());
            newResume.setSkillIds(newSkillList);
            newResume.getSkillIds().remove(Integer.valueOf(id));
            
            setResume(resume, newResume);
```